test.js

```diff
@@ -102,4 +103,36 @@ describe('dat-gateway', function () {
       throw e
     })
   })
+
+  it('should open a websocket to /peers', function () {
+    const peers = `ws://localhost:5917/peers`
+    let socket = null
+    return new Promise((resolve, reject) => {
```

Comment:
This `new Promise` thing isn't necessary anymore since everything in the body is already using promises.

Get rid of it (and the console.* calls) and have `return axios.get( etc..`. You can also get rid of the try catch block.

Reply:
Almost there, have an issue w/ the after-all hook: 

```
    ✓ should open a websocket to /peers (3186ms)
    1) "after all" hook


  6 passing (24s)
  1 failing

  1) dat-gateway
       "after all" hook:
     Uncaught Error: connect ECONNREFUSED 127.0.0.1:5917
      at Object._errnoException (util.js:1003:13)
      at _exceptionWithHostPort (util.js:1024:20)
      at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1195:14)
```

looking at maxogden's test-client.js...

Reply:
Is this test bringing down the server? I did add a `socket.end()` inside the socket.on, but when I remove it, there error `ECONNREFUSED 127.0.0.1:5917` is still thrown.